### PR TITLE
Check if given number is even or odd

### PR DESCRIPTION
**Pull request by ra7kar**

### CODE REVIEW

File: data_structures/is_even.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Find if a number is even or odd,
+# without using the mod, div or multiplication operators.
+
+
+def is_even(n):
```
Comment (atrifex) [inline data_structures/is_even.py:5]: No, this implementation is way, way too slow. You can do this in O(n).

Look into `&` bit operator and think about what bit to check to see if a number is even.

Comment (ra7kar) [reply data_structures/is_even.py:5]: There are three ways to optimize this code, I have tried the Bit wise operation AND, OR and XOR. All three were successful.

File: data_structures/is_even.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Find if a number is even or odd,
+# without using the mod, div or multiplication operators.
+
+
+def is_even(n, option="AND"):
```
Comment (atrifex) [inline data_structures/is_even.py:5]: Use Enum to decide on options not strings.

Comment (ra7kar) [reply data_structures/is_even.py]: Enum used instead of strings

File: data_structures/is_even.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    AND = 0
+    OR = 1
+    XOR = 2
+
+
+def is_even(n, option):
+
+    # using Bit wise AND operation
+    if option == Operations.AND:
+        if n & 1 == 1:
```
Comment (atrifex) [inline data_structures/is_even.py:17]:
```suggestion
        return not (n & 1 == 1)
```

File: data_structures/is_even.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            return False  # ODD
+        return True  # EVEN
+
+    # using bit wise OR operation
+    if option == Operations.OR:
+        if n | 1 > n:
```
Comment (atrifex) [inline data_structures/is_even.py:23]:
```suggestion
        return n | 1 > n
```

**Pull request merged**